src/Rapidez.php

```diff
@@ -96,6 +96,17 @@ public function setStore(Store|array|callable|int|string $store): void
         config()->set('rapidez.store_code', $store['code']);
         config()->set('rapidez.website', $store['website_id']);
         config()->set('rapidez.website_code', $store['website_code']);
+        config()->set('rapidez.group', $store['group_id']);
         config()->set('rapidez.root_category_id', $store['root_category_id']);
     }
+
+    public function withStore(Store|array|callable|int|string $store, callable $callback)
```

Comment:
I think `withStore` is a bit of an ambiguous name, `asStore` may be a better name.
Maybe it's nice to allow arguments to be passed in as well.

```suggestion
    public function asStore(Store|array|callable|int|string $store, callable $callback, ...$args)
```
```php
$result = $callback(...$args);
```